src/client/Microsoft.Identity.Client/Platforms/iOS/iOSBroker.cs

```diff
@@ -72,6 +85,11 @@ private void AddIosSpecificParametersToPayload(Dictionary<string, string> broker
             brokerPayload[iOSBrokerConstants.BrokerKey] = encodedBrokerKey;
             brokerPayload[iOSBrokerConstants.MsgProtocolVer] = BrokerParameter.MsgProtocolVersion3;
 
+            if (!string.IsNullOrEmpty(BrokerRequestNonce))
```

Comment:
No, it would be better to check if the V3 broker can be used and to generate the nonce here.